### PR TITLE
Use OS Mbean instead of reflection and enhance Linux container memory calculation

### PR DESCRIPTION
**Pull request by kgibm**

### CODE REVIEW

**Comment by kgibm** [comment]
#build

**Comment by kgibm** [comment]
Personal build has one unrelated error in com.ibm.ws.security.authentication.fat.AuthCacheTest

File: dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/util/MemoryInformation.java [hunk 1]
```diff
@@ -419,13 +560,12 @@ private long applySwappiness(long cached) throws IOException, MemoryInformationE
...
-            List<String> swappinessLines = FileSystem.readAllLines("/proc/sys/vm/swappiness");
-            if (swappinessLines.size() == 1) {
-                return Integer.parseInt(swappinessLines.get(0));
-            } else {
-                throw new MemoryInformationException(MessageFormat.format(BootstrapConstants.messages.getString("memory.information.unexpected"), "/proc/sys/vm/swappiness",
-                                                                          swappinessLines));
+            // Use cgroup swappiness if available
+            String fileToRead = "/proc/sys/vm/swappiness";
+            if (FileSystem.fileExists("/sys/fs/cgroup/memory/memory.swappiness")) {
+                fileToRead = "/proc/sys/vm/swappiness";
```
Comment (gjdeval) [inline dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/util/MemoryInformation.java]: Should this refer to the cgroup swappiness file instead?

Comment (kgibm) [reply dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/util/MemoryInformation.java]: Great catch!

**Comment by kgibm** [comment]
#build

**Pull request merged**